Fix sea.js issues #962, #957, #956

mimiza opened the pull request:
Fixes these issues:
(https://github.com/amark/gun/issues/956)
(https://github.com/amark/gun/issues/957)
(https://github.com/amark/gun/issues/962)
BEFORE:
- user.recall() uses alias/pass -> not reliable, only works with auth(alias,pass), doesn't work with auth(pair)
- user.auth(pair, cb) -> cb is not a function

AFTER:
- user.recall() uses pair -> more reliable, works with auth(pair)
- user.auth(pair, cb) -> works

capitalistdog commented:
This appears to have resolved the initial issue in our tests, but we're investigating issues with overwriting existing user data (unsure if related still)

amark commented:
Yaay!

@mimiza @capitalistdog @jabis can we do a call today/tomorrow or Wed+ to review & merge? I think jabis saw a couple issues that might break some apps? So hopefully on call we can work that out & move to publish.

mimiza commented:
We all have different timezones. But yeah, please arrange a call, but please let me know what time. Or we can just discuss here, it's ok for me. My English listening skill is not so fluent, just warning.

mimiza commented:
I have just switched to ancient JS technique, so it would be more compatible with those stone-age browsers. Removed (...args), replaced with 'arguments' :D

amark commented:
@mimiza yay!

Timezones I've found early CA often works for everybody?

@mimiza @capitalistdog @jabis Wednesday 10am?

mimiza commented:
By CA do you mean California (GMT-7) timezone? Mine is GMT+7 which is 12 hours foward. So your 10AM is my 10PM. Ok Wed 10PM.

mimiza commented:
@amark Oh but where do we meet? Meething I guess? Somewhere else?

sirpy commented on sea.js:
```diff
@@ -825,16 +825,21 @@
     // now that we have created a user, we want to authenticate them!
-    User.prototype.auth = function(alias, pass, cb, opt){
+    User.prototype.auth = function(){
+      const alias = typeof arguments[0] === 'string' ? arguments[0] : null
```
i'd keep arguments as part of the function signature, otherwise its not very readable or dev friendly

amark replied:
yes, let's refactor it.

amark commented:
@jabis @mimiza @sirpy @capitalistdog + whoever else:

I'll be on:

https://us.meething.space/?room=gun

in about 15min from now. Is it still a good time?

amark commented:
we switched to this: https://whereby.com/danlugo92

amark commented:
w0000000h00000000 🍾 🎈 🎉 🎈 🎉 🎈 🎉 🎈 🎉  you are an AWESOME ROLE MODEL for OPEN SOURCE :D :D :D

mimiza commented:
Thank you all ❤️😊
